### PR TITLE
ipq40xx: improve support for Edgecore ECW5211

### PR DESCRIPTION
**Pull request by mans0n**
* Drop the local BDFs as those are already in the upstream under different names
* Sort 01_leds alphabetically
* Use label for the mdio node
* Drop the unused usb node labels
* Make uboot-env partition writable
* Tidy up the device recipe

### CODE REVIEW

**Comment by mans0n** [comment]
```
$ python2 ath10k-bdencoder -i board-2.bin
...
BoardNames[2]: 'bus=ahb,bmi-chip-id=0,bmi-board-id=18'
BoardLength[2]: 12064
BoardCRC32[2]: 628eba64
BoardMD5[2]: 11c70b64c714c30d040fcd7b0ef3f53f
...
BoardNames[39]: 'bus=ahb,bmi-chip-id=0,bmi-board-id=25,variant=ALFA-Network-AP120C-AC'
BoardLength[39]: 12064
BoardCRC32[39]: b9121b69
BoardMD5[39]: e676f159de8afd74df3fecc8ecf94389
...

$ python2 ath10k-bdencoder -i board-edgecore_ecw5211.qca4019
...
BoardNames[0]: 'bus=ahb,bmi-chip-id=0,bmi-board-id=18,variant=Edgecore-ECW5211'
BoardLength[0]: 12064
BoardCRC32[0]: 628eba64
BoardMD5[0]: 11c70b64c714c30d040fcd7b0ef3f53f
BoardNames[1]: 'bus=ahb,bmi-chip-id=0,bmi-board-id=25,variant=Edgecore-ECW5211'
BoardLength[1]: 12064
BoardCRC32[1]: b9121b69
BoardMD5[1]: e676f159de8afd74df3fecc8ecf94389
```

**Comment by mans0n** [comment]
@robimarko Hi, do you still have access to the Edgecore ECW5211 device? I'd like to ask you a few questions before adding more changes.

**Comment by adschm** [comment]
Please add "Fixes: aaaaaabbbbbb ("name")" to the commit message.

File: target/linux/ipq40xx/image/Makefile [hunk 1]
```diff
@@ -347,24 +347,26 @@ TARGET_DEVICES += dlink_dap-2610
...
+	DEVICE_PACKAGES := kmod-tpm-i2c-atmel kmod-usb-acm uboot-envtools
 endef
 TARGET_DEVICES += edgecore_ecw5211
 
 define Device/edgecore_oap100
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
-	DEVICE_TITLE := Edgecore OAP100
+	DEVICE_VENDOR := Edgecore
+	DEVICE_MODEL := OAP100
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:364]: I'd keep the OAP100 changes separate.

Comment (mans0n) [reply target/linux/ipq40xx/image/Makefile:364]: Sure. This is still a draft, I'll clean it up in the next round.

**Comment by robimarko** [comment]
> @robimarko Hi, do you still have access to the Edgecore ECW5211 device? I'd like to ask you a few questions before adding more changes.

Yes, feel free to ask

**Comment by mans0n** [comment]
Thanks. :)

0. I also have a device, but its precise model name is ECW5211-L. Does yours end with "-L" or not?
1. Is there a reason you chose to switch two gmacs? To me it feels more convenient to use eth0 as lan and eth1 as wan.
2. Would you mind if I add `// SPDX-License-Identifier: GPL-2.0-or-later OR MIT` to the dts?
3. Does it store MAC addresses in the ART partition?
4. What happens if we omit `DEVICE_DTS_CONFIG` in the image Makefile?
5. I'm wondering if extending the rootfs to 128MB is worth enough to force users to modify uboot env... What do you think?

**Comment by robimarko** [comment]
Hi,
0. Yes, its the -L version(I don't think any other exist)
1. Requirement of the client was to keep the config as close as possible to the stock FW
2. No, feel free as I forgot it.
3. Yes, you can clearly see 2 MAC addresses at 0x0 of ART, that is really common
4. Nothing as long as SOC variable is set correctly
5. Not really, that requires too invasive changes for the average users and you would loose the redundancy.

**Comment by mans0n** [comment]
Thanks for your explanation.

> Requirement of the client was to keep the config as close as possible to the stock FW

Okay, that makes sense.

> Yes, you can clearly see 2 MAC addresses at 0x0 of ART, that is really common

At first I thought we need to specify `mtd-mac-address` but I've just realized that the bootloader automatically replaces `local-mac-address` with correct ones.

> Nothing as long as SOC variable is set correctly

I guess you included it to follow the stock FW behavior, but openwrt FW fdt has only one config entry (unlike stock FW), so I don't think the name does not matter much here. Please let me know if you disagree.

> - USB: 1 x USB 3.0 SuperSpeed

Really? The product page says ECW5211-L only supports USB 2.0.

I added some more changes, please have a look.

**Comment by adschm** [comment]
@mans0n I assume I can already pick the OAP100 commit?

**Comment by mans0n** [comment]
@adschm Sure, that would be nice.

**Comment by adschm** [comment]
Picked the OAP100 for now.

Will take the other one when you state it's ready. Since you "change" the license, I will need an Acked-by by Robert in any case.

**Comment by mans0n** [comment]
@robimarko May I ask you for a quick review before this get merged?

**Comment by robimarko** [comment]
I have only one comment, and that is why not set WLAN LED-s via DTS?
Triggers can be set there instead of using the userspace script.

Also, can you rebase this onto master as your OAP commit got already merged?

**Comment by pepe2k** [comment]
@mans0n 

> It seems AP120C-AC and ECW5211 have exactly the same board according to the FCC internal photos.

I can confirm that, both `vendors` use the same board manufactured by Accton. The only differences are software and 256 MB or RAM in AP120C-AC (vs. 128 MB in the EdgeCore - at least that was true at the time I added support for ALFA board).

**Comment by mans0n** [comment]
> why not set WLAN LED-s via DTS?

Oh, I thought that was your intention... Moved it now.

> No, feel free as I forgot it.

I translated that into Acked-by.

@pepe2k Thanks for the confirmation. Now I realized that Edgecore is a subsidiary of Accton...

**Comment by mans0n** [comment]
Rebased due to ad90a42e4875. I think it is ready for merge now.

**Comment by adschm** [comment]
Thanks, merged.

**Pull request closed without merging**